src/Stratis.Bitcoin/P2P/Peer/NetworkPeerClient.cs

```diff
@@ -145,6 +154,148 @@ public NetworkPeerClient(int Id, TcpClient tcpClient, ILoggerFactory loggerFacto
             this.logger.LogTrace("(-)");
         }
 
+        /// <summary>
+        /// Reads raw message in binary form from the connection stream.
+        /// </summary>
+        /// <param name="protocolVersion">Version of the protocol that defines the message format.</param>
+        /// <param name="cancellation">Cancellation token that allows aborting the read operation.</param>
+        /// <returns>Binary message received from the connected counterparty.</returns>
+        /// <exception cref="OperationCanceledException">Thrown if the operation was cancelled or the end of the stream was reached.</exception>
+        /// <exception cref="FormatException">Thrown if the incoming message is too big.</exception>
+        public async Task<byte[]> ReadMessageAsync(ProtocolVersion protocolVersion, CancellationToken cancellation = default(CancellationToken))
```

Comment:
I would suggest we remove the `default(CancellationToken)` to avoid scenarios where we miss use it, unless you see a scenario where we would not use cancellation when reading from the socket.

Reply:
Actually I take this back leave the cancellation here.

Reply:
ok, i remove it in later PR